ssl/ssl_privkey.cc

```diff
@@ -368,28 +368,69 @@ enum ssl_private_key_result_t ssl_private_key_decrypt(SSL_HANDSHAKE *hs,
   return ssl_private_key_success;
 }
 
-bool ssl_private_key_supports_signature_algorithm(SSL_HANDSHAKE *hs,
-                                                  uint16_t sigalg) {
-  SSL *const ssl = hs->ssl;
-  if (!pkey_supports_algorithm(ssl, hs->local_pubkey.get(), sigalg)) {
-    return false;
-  }
-
+static bool ssl_public_key_rsa_pss_check(EVP_PKEY *pubkey, uint16_t sigalg) {
   // Ensure the RSA key is large enough for the hash. RSASSA-PSS requires that
   // emLen be at least hLen + sLen + 2. Both hLen and sLen are the size of the
   // hash in TLS. Reasonable RSA key sizes are large enough for the largest
   // defined RSASSA-PSS algorithm, but 1024-bit RSA is slightly too small for
   // SHA-512. 1024-bit RSA is sometimes used for test credentials, so check the
   // size so that we can fall back to another algorithm in that case.
   const SSL_SIGNATURE_ALGORITHM *alg = get_signature_algorithm(sigalg);
-  if (alg->is_rsa_pss && (size_t)EVP_PKEY_size(hs->local_pubkey.get()) <
-                             2 * EVP_MD_size(alg->digest_func()) + 2) {
+  if (alg->is_rsa_pss &&
+      (size_t)EVP_PKEY_size(pubkey) < 2 * EVP_MD_size(alg->digest_func()) + 2) {
```

Comment:
Should this be less than or equal to?

Reply:
From the comment above, "at least hLen + sLen + 2" makes it seems like `hLen + sLen + 2` is a sufficient size to be used.
Just looked into OpenSSL's code and the same check is also enforced :) 
https://github.com/openssl/openssl/blob/master/ssl/t1_lib.c#L1628-L1636